src/main/java/gov/nasa/jpl/mbee/mdk/mms/validation/DiagramValidator.java

```diff
@@ -103,53 +107,26 @@ public void run(ProgressStatus progressStatus) {
                     if ((idNode = jsonNode.get(MDKConstants.ID_KEY)) == null || !idNode.isTextual()) {
                         continue;
                     }
-                    diagramElementsMap.put(idNode.asText(), (ObjectNode) jsonNode);
-                    Set<String> artifactIds = elementArtifactMap.computeIfAbsent(idNode.asText(), o -> new LinkedHashSet<>());
-                    JsonNode artifactsNode;
-                    if ((artifactsNode = jsonNode.get(MDKConstants.ARTIFACT_IDS_KEY)) != null && artifactsNode.isArray()) {
-                        artifactsNode.forEach(node -> artifactIds.add(node.asText()));
+                    Set<String> artifactExtensions = elementArtifactMap.computeIfAbsent(idNode.asText(), o -> new LinkedHashSet<>());
+                    JsonNode artifactsArray;
+                    if ((artifactsArray = jsonNode.get(MDKConstants.ARTIFACTS_KEY)) != null && artifactsArray.isArray()) {
+                        artifactsArray.forEach(node -> {
+                            artifactExtensions.add(idNode.asText() + '_' + node.get("extension").asText());
+                            artifactsMap.put(idNode.asText() + '_' + node.get("extension").asText(), (ObjectNode) node);
```

Comment:
Wouldn't hurt to check that node is indeed an ObjectNode before casting. Server can always send unexpected data.